.github/workflows/push.yaml

```diff
@@ -20,13 +20,29 @@ jobs:
         python-version: ["3.10", "3.11", "3.12"]
     steps:
       - uses: actions/checkout@v4
-      - name: Setup Python.
+      - name: Setup Python
         uses: actions/setup-python@v5
         with:
           python-version: ${{matrix.python-version}}
-      - name: Install dependencies.
+      - name: Install uv
         run: |
-          pip install poetry
-          poetry install --with dev
-      - name: Run tests.
-        run: poetry run pytest tests
+          pip install uv
+      - name: Install dependencies
+        shell: bash
+        run: |
+          uv venv
+          if [[ "$RUNNER_OS" == "Windows" ]]; then
+            source .venv/Scripts/activate
+          else
+            source .venv/bin/activate
```

Comment:
let uv manage the venv you shouldn't need to activate it here

Reply:
If we remove this, the test would fail
<img width="402" alt="截屏2024-12-25 23 28 55" src="https://github.com/user-attachments/assets/651c2e4e-b2ec-4512-9079-ea80479c6a2b" />

Reply:
Yes there were a few issues with the github action which were causing it to fail (see the other comment thread) but you definitely don't want to manually source a virtual environment like you have here.

The point of uv is that it manages to virtual environment for you, running `uv run ...` will execute in a virtual environment. Trying to put yourself into a venv beforehand will just make the state of your environment more confusing for you.

I'd recommend reading the uv guide to build a better mental model of what the tool is doing. :)